posthog/batch_exports/http.py

```diff
@@ -425,6 +425,15 @@ def unpause(self, request: request.Request, *args, **kwargs) -> response.Respons
 
         return response.Response({"paused": False})
 
+    @action(methods=["POST"], detail=True)
+    def archive(self, request: request.Request, *args, **kwargs) -> response.Response:
+        """Archive a BatchExport."""
+        if not isinstance(request.user, User) or request.user.current_team is None:
+            raise NotAuthenticated()
+
+        disable_and_delete_export(self.get_object())
+        return response.Response({"archived": True})
```

Comment:
This is already covered by the perform_destroy method below, why not call that instead?

Reply:
Also, the auth check should not be needed: https://github.com/PostHog/posthog/pull/21724/files#r1574733513

Reply:
We could use that, the reason I didn't is because I wasn't sure we want users to be able to click on buttons and delete stuff from PG, vs just marking it as "deleted" so we have the info if needed, e.g. for debugging why a batch export caused problems for the system as a whole

Reply:
But this action is _exactly the same_ as `perform_destroy` aka HTTP DELETE, so it's quite pointless, in addition to feeling unRESTful. In its current shape, it doesn't have a reason for existence

Reply:
I don't have strong feelings about this. I figured we'd see a support ticket for "I accidentally deleted my export, can I get it back" and if we use hard delete we can't give it back, unless I'm misunderstanding how `perform_destroy` works. That said I'm fine with just using hard delete, no strong feelings here.

Reply:
I mean, both `archive` and `perform_destroy` call `disable_and_delete_export()` – neither does anything else. They are one and the same, both soft-delete

Reply:
My potentially wrong understanding of the world is:
1. if I send `/batch_export/ABC/delete` <- then `destroy` is called, which calls `perform_destroy` and finally data from PG is deleted
2. if I send `/batch_export/ABC/archive` <- then `disable_and_delete_export()` is called and the object in PG has a column `deleted=True`, but still exists in PG.

Reply:
Ah, your understanding would be correct if we had `super().perform_destroy(instance)` in there, because `perform_destroy` _normally_ deletes data from PG. But in _this_ case it's overridden completely and these mechanics are fully replaced with `disable_and_delete_export`

Reply:
aah ok thanks, I wasn't sure how django magic works ... clearly 😅